### PR TITLE
 :sparkles: Convert dtypes to safe types Float64, Int64 and string

### PR DESCRIPTION
**Pull request by Marigold**
Implements https://github.com/owid/etl/issues/3277

Adds new parameter `ds.read_table("my_table", safe_types=True)` that converts all types to "safe" types `Float64`, `Int64` or `string[python]`. It's true by default.

(We've also considered the name `unpack` in the past instead of `safe_types`. I don't have a preference.)

## int64 vs Int64

Should we use numpy's `int64` type or nullable pandas type `Int64` (or even pyarrow types)? This PR goes with nullable types, but there are some hidden risks to it:
- Mixing `np.nan` and `pd.NA` creates a nightmare of compatibility issues. We should always use `pd.NA`
- Sometimes `np.nan` sneaks in, for instance division 0/0 returns `np.nan` even if we work with nullable types
```python
a = pd.Series([0]).astype("Int64")
b = pd.Series([0]).astype("Int64")
# the first item is np.nan, not pd.NA!
c = a / b
# this is actually False, `pd.isnull` doesn't detect np.nan when dtype is Int64
pd.isnull(c)
# this doesn't work either...
c.fillna(pd.NA)
# only this fixes it
c.mask(np.isnan(c), pd.NA)
```

## TODO before merging:
- [x] Increment ETL_EPOCH and make sure old datasets don't change
- [x] Undo ETL_EPOCH increment

### CODE REVIEW

File: lib/repack/tests/test_repack.py [hunk 1]
```diff
@@ -253,3 +253,73 @@ def test_repack_string_type():
...
+    assert df_safe.index.levels[1].dtype == "string[python]"  # type: ignore
+    # Check column dtype
+    assert df_safe["float_col"].dtype == "Float64"
+
+
+def test_to_safe_types_with_nan():
+    # Create a DataFrame with NaN values
+    df = pd.DataFrame(
+        {
+            "int_col": [1, 2, 3],
```
Comment (pabloarosado) [inline lib/repack/tests/test_repack.py:308]: It seems that is_integer_dtype returns False if it's a series of integers with nans. So, it's testing for `int`, not for `Int64`. I suppose that, ideally, it should return True, and we would load all integer columns (with possible nans) as `Int64`.
Should we include that extra logic? I'm not sure about it.

**Review by pabloarosado — APPROVED** [review]
Thanks for doing this! Changing types after repacking is a common headache.
I'm a bit unsure of the preferred dtypes. Maybe we should discuss them on the data architecture call (I agree with your choices, but I'm not sure if there may be some negative side effects).

**Comment by Marigold** [comment]
I've updated it to use nullable types & string[pyarrow]. The first commit contains changes to steps that are not worth checking (datadiff confirmed they match), so please only look at [changes from the second commit](https://github.com/owid/etl/pull/3474/files/6cb9bfdb2dbe8305df5919d9ca1c2f9b6021a9ce..HEAD).

## Changes
- Repack library converts everything to **nullable types and categoricals** rather than a mix of nullable types, numpy types and categoricals
- New function `to_safe_types` that converts to `Int64`, `Float64` and `string[pyarrow]` types
- New method `ds.read(...)` (it used to be called `read_table`) that reads a table and by default resets index and converts to safe types
- Reading data from snapshot with `snap.read(...)` also uses safe types by default
- Update pandas to 2.2.3 (fixes problems with pyarrow)
- Division 0/0 returns `pd.NA` when working with Tables (it doesn't cover edge cases, but if they happen, you'd get an error when saving a dataset)

**Comment by Marigold** [comment]
@pabloarosado This is finally ready for review. Check the comment above for a summary of changes.

I wanted to go one step further and enable `pd.options.future.infer_string = True`, which makes the new `string[pyarrow]` the default. I ran into some issues, though, so I'm leaving it for either a future PR or for the actual upgrade to pandas 3.

**Review by pabloarosado — APPROVED** [review]
Huge, thanks for this massive refactor! I have quickly scanned through all files and it all looks good. And you already checked for data changes. So I think this is ready to go!

**Pull request merged**